CRM/Activity/Selector/Search.php

```diff
@@ -274,7 +274,7 @@ public function &getRows($action, $offset, $rowCount, $sort, $output = NULL) {
     $assigneeID = CRM_Utils_Array::key('Activity Assignees', $activityContacts);
     $targetID = CRM_Utils_Array::key('Activity Targets', $activityContacts);
     // Get all activity types
-    $activityTypes = CRM_Core_PseudoConstant::activityType(TRUE, TRUE, FALSE, 'name', TRUE);
+    $activityTypes = CRM_Activity_BAO_Activity::buildOptions('activity_type_id', 'validate');
```

Comment:
This is correct but it would be better just to do 
```
    $bulkActivityTypeID = CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_type_id', 'Bulk Email');
```

& not use this $activityTypes param at all

Reply:
Yes, done